pkg/istio-agent/xds_proxy.go

```diff
@@ -119,14 +123,57 @@ func initXdsProxy(ia *Agent) (*XdsProxy, error) {
 		LocalHostAddr: localHostAddr,
 	}
 	proxy := &XdsProxy{
-		istiodAddress:  ia.proxyConfig.DiscoveryAddress,
-		clusterID:      ia.secOpts.ClusterID,
-		localDNSServer: ia.localDNSServer,
-		stopChan:       make(chan struct{}),
-		healthChecker:  health.NewWorkloadHealthChecker(ia.proxyConfig.ReadinessProbe, envoyProbe),
-		xdsHeaders:     ia.cfg.XDSHeaders,
-		xdsUdsPath:     ia.cfg.XdsUdsPath,
-		wasmCache:      wasm.NewLocalFileCache(constants.IstioDataDir, wasm.DefaultWasmModulePurgeInteval, wasm.DefaultWasmModuleExpiry),
+		istiodAddress: ia.proxyConfig.DiscoveryAddress,
+		clusterID:     ia.secOpts.ClusterID,
+		handlers:      map[string]ResponseHandler{},
+		stopChan:      make(chan struct{}),
+		healthChecker: health.NewWorkloadHealthChecker(ia.proxyConfig.ReadinessProbe, envoyProbe),
+		xdsHeaders:    ia.cfg.XDSHeaders,
+		xdsUdsPath:    ia.cfg.XdsUdsPath,
+		wasmCache:     wasm.NewLocalFileCache(constants.IstioDataDir, wasm.DefaultWasmModulePurgeInteval, wasm.DefaultWasmModuleExpiry),
+	}
+
+	if ia.localDNSServer != nil {
+		proxy.handlers[v3.NameTableType] = func(resp *discovery.DiscoveryResponse) error {
+			if ia.localDNSServer == nil {
+				return nil
+			}
+			if len(resp.Resources) == 0 {
+				return fmt.Errorf("empty response")
+			}
+			var nt nds.NameTable
+			if err := ptypes.UnmarshalAny(resp.Resources[0], &nt); err != nil {
+				log.Errorf("failed to unmarshall name table: %v", err)
+				return err
+			}
+			ia.localDNSServer.UpdateLookupTable(&nt)
+			return nil
+		}
+	}
+	if ia.secretCache != nil { // TODO: add a config option for this
```

Comment:
@howardjohn - this was originally your comment. Not sure what configuration option you were considering?

Reply:
Why would secretCache be nil ?